### DISCUSSION TITLE
Plex Shaders on Clients

### DISCUSSION THREAD

**Discussion by 4RK0S4H4** [bloc97/Anime4K#243] [Q&A]
Hello, I saw that instructions were added to add anime4K to Plex. I was just curious if it applied it at the server level to where the upscaled version was streamed to other clients like my Apple TV, or if it was more meant to upscale on just that specific machine's client where I place the shaders?

**Comment by bloc97** [comment]
If I remember correctly, it upscales on the client's machine using the local GPU, and not on the server that serves the files.

**Reply by adamlogan** [reply]
It would be amazing if this could be done on the server, although I have to admit my server isn't up to the task. Only my M1 Pro MBP has enough GPU to do this.

**Reply by Tama47** [reply]
@adamlogan There was this [custom build of Plex Transcoder that supports on-the-fly mpv filters like Anime4K](https://www.reddit.com/r/PleX/s/ebbNFIap4H), but it never got anywhere. Unless someone could revive the project.

Alternatively, since you have an M1 Pro, you could pre-upscale videos beforehand and put them on your Plex server. I recommend the [Waifu2x app on the App Store](https://apps.apple.com/app/id1286485858); it has the Anime4K shaders that uses Core ML and is super fast.

Last option, and probably the easiest, pair your Apple TV remote to your MacBook and plug it into the TV via HDMI. The Plex HTPC (for macOS Home Theater PCs) app functions exactly like a TV app.
